test/db/cmd/types

```diff
@@ -376,6 +376,7 @@ EOF
 RUN
 
 NAME=typedef self-reference
+BROKEN=1
```

Comment:
Have you tried to fix this? As I noted before, it turned out that many DWARF files store typedefs in the similar chain, so it breaks many files actually.

Reply:
I remember the problem I had was that typedb uses typename as the identifier, but sometimes there are different types with the same typename.

Reply:
The problem is that I thought it's not common, but it turned out most files I tried to open suddently show `unknown_t` instead of valid types because it first defines `typedef` to some type, e.g. `struct`, and only then defines this `struct`. I think we should allow such forward definitions somehow. You can create additional hashtable of these forward definitions and check it - this is how I handle the same situation in the C parser.

Reply:
I think that pattern is very common indeed.

Reply:
> Have you tried to fix this? As I noted before, it turned out that many DWARF files store typedefs in the similar chain, so it breaks many files actually.

Can you help test this issue using the latest code?